api/src/policies/notifications-policy.ts

```diff
@@ -0,0 +1,45 @@
+import { Attributes, FindOptions } from "@sequelize/core"
+
+import { Path } from "@/utils/deep-pick"
+import { User, Notification } from "@/models"
+import { PolicyFactory } from "@/policies/base-policy"
+
+export class NotificationsPolicy extends PolicyFactory(Notification) {
+  show(): boolean {
+    if (this.record.userId === this.user.id) return true
+
+    return false
+  }
+
+  create(): boolean {
+    if (this.user.isSystemAdmin) return true
+
+    return false
+  }
+
+  update(): boolean {
+    if (this.record.userId === this.user.id) return true
+
+    return false
+  }
+
+  destroy(): boolean {
+    if (this.user.isSystemAdmin) return true
+
+    return false
+  }
```

Comment:
I would remove this access as well. Since notifications are between the user and the system, I think only the user and the system should be able to modify them.